main/org.thingpedia.iot.climate/dataset.tt

```diff
@@ -457,53 +432,28 @@ dataset @org.thingpedia.iot.climate language "en" {
                    "set to diffuse the ${p_name:const} fan",
                    "switch ${p_name:const} fan to diffuse"]];
 
-    action (p_name : String, f_mode :Enum(on,off,auto,low,medium,high,middle,focus,diffuse)) := @org.thingpedia.iot.climate(name=p_name).set_swing_mode(mode=f_mode)
-    #_[utterances=["set the ${p_name:const} swing to $f_mode"]];
-
-    action (p_name : String) := @org.thingpedia.iot.climate(name=p_name).set_swing_mode(mode=enum(on)) 
-    #_[utterances=["turn the ${p_name:const} swing on",
-                   "set to on the ${p_name:const} swing",
-                   "switch ${p_name:const} swing on"]];
+    action (p_name : String, f_mode :Enum(off,vertical,horizontal,both)) := @org.thingpedia.iot.climate(name=p_name).set_swing_mode(mode=f_mode)
```

Comment:
Use `p_mode` here. By convention, parameters to primitive templates start with `p_`, and that's important because we use the parameter name to generate a placeholder phrase.
In this case, we generate "set the ${p_name} swing to some mode" if you use `p_mode`, but we generate "set the ${p_name} swing to some f mode" if you use `f_mode`. The latter is worse.